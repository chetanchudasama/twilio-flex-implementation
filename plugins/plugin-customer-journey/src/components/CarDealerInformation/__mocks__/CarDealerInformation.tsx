import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
