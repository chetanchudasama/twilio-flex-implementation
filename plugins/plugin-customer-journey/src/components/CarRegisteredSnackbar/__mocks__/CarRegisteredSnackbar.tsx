import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
