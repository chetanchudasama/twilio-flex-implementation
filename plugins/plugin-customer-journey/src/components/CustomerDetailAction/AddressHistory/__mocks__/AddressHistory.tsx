import React from "react";
import { AddressHistoryProps } from "../AddressHistory";

const AddressHistory: React.FC<AddressHistoryProps> = (props) => (
  <div className="address-history" />
);

export default AddressHistory;
