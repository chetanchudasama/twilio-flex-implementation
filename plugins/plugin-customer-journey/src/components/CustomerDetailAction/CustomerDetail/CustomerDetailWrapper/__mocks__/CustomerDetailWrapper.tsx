import React from "react";
import { ComponentProps } from "../CustomerDetailWrapper";

const CustomerDetailWrapper: React.FC<ComponentProps> = (props) => (
  <div className="customer-detail-wrapper" />
);

export default CustomerDetailWrapper;
