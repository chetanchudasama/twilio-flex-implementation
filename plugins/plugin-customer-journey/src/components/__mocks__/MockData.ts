import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
