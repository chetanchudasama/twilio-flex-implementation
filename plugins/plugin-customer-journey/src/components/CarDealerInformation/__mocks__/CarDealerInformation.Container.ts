import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
