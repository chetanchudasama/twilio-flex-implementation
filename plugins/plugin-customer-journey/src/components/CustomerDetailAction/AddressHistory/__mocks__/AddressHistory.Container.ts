import AddressHistory from "./AddressHistory";

export default AddressHistory;
