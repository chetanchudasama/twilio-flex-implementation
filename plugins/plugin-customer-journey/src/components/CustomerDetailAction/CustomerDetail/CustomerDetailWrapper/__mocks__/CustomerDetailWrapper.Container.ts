import CustomerDetailWrapper from "./CustomerDetailWrapper";

export default CustomerDetailWrapper;
