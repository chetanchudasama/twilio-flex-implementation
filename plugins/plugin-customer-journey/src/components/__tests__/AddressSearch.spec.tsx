import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import { Button } from "@material-ui/core";
import { Manager } from "@twilio/flex-ui";
import { StyledLinkButton } from "@common/components";
import {
  AddressSearch,
  AddressSearchProps,
} from "../AddressSearch/AddressSearch";
import { flexManagerMock } from "../__mocks__/MockData";

Manager.getInstance = flexManagerMock;

const mockUpdate = jest.fn();

const props: AddressSearchProps = {
  updateAddress: mockUpdate,
  address: {
    buildingName: "test-building-name",
    buildingNumber: "1",
    postcode: "POST CODE 1",
    streetName: "Side Street",
    subBuilding: "31A sub-building",
    town: "Fictional Town",
  },
};

describe("AddressSearch", () => {
  afterEach(() => {
    mockUpdate.mockClear();
    jest.clearAllTimers();
    jest.clearAllMocks();
  });

  it("renders correctly", async () => {
    expect(
      create(
        <AddressSearch
          updateAddress={props.updateAddress}
          address={props.address}
        />
      ).toJSON()
    ).toMatchSnapshot();
  });

  it("calls updateAddress, on value change in each input field", () => {
    const wrapper = mount(
      <AddressSearch updateAddress={props.updateAddress} />
    );

    wrapper.find(StyledLinkButton).at(0).simulate("click");
    wrapper.update();

    const subBuildingInput = wrapper.find("input").at(0);
    subBuildingInput.simulate("change", { target: { value: "subBuilding" } });

    const buildingNumber = wrapper.find("input").at(1);
    buildingNumber.simulate("change", { target: { value: "5" } });

    const buildingName = wrapper.find("input").at(2);
    buildingName.simulate("change", { target: { value: "buildingName" } });

    const streetName = wrapper.find("input").at(3);
    streetName.simulate("change", { target: { value: "streetName" } });

    const townName = wrapper.find("input").at(4);
    townName.simulate("change", { target: { value: "town" } });

    const postcode = wrapper.find("input").at(5);
    postcode.simulate("change", { target: { value: "9WZ" } });

    expect(mockUpdate).toBeCalledTimes(6);
  });

  it("is in readonly state when address props are passed in, then goes to edit mode when edit is clicked", async () => {
    const wrapper = mount(
      <AddressSearch
        updateAddress={props.updateAddress}
        address={props.address}
      />
    );

    wrapper.update();
    expect(
      wrapper.find({ id: "address-readonly-town-name" }).at(0).text()
    ).toBe(props.address?.town);
    expect(
      wrapper.find({ id: "address-readonly-street-name" }).at(0).text()
    ).toBe(props.address?.streetName);
    expect(
      wrapper.find(Button).find({ id: "address-readonly-edit" }).at(0).exists()
    ).toBeTruthy();
    wrapper
      .find(Button)
      .find({ id: "address-readonly-edit" })
      .at(0)
      .simulate("click");
    wrapper.update();
    expect(
      wrapper.find(Button).find({ id: "address-readonly-edit" }).at(0).exists()
    ).toBeFalsy();
  });
});
