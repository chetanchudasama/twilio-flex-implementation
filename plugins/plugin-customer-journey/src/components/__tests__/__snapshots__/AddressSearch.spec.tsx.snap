// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressSearch renders correctly 1`] = `
<div
  className="css-1e54dbh"
>
  <div
    className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41 address-configuration"
  >
    <div
      className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41 address-current"
    >
      <div
        className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-10-39"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          31A sub-building, 
          test-building-name, 
          1
        </div>
        <div
          className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
          id="address-readonly-street-name"
        >
          Side Street
        </div>
        <div
          className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
          id="address-readonly-town-name"
        >
          Fictional Town
        </div>
        <div
          className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
        >
          POST CODE 1
        </div>
      </div>
      <div
        className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-align-items-xs-flex-end-11 MuiGrid-align-content-xs-flex-start-14 MuiGrid-justify-xs-flex-end-19 MuiGrid-grid-xs-2-31"
      >
        <button
          className="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-text-100 MuiButton-textSecondary-102 MuiButton-flat-103 MuiButton-flatSecondary-105 address-edit-button"
          disabled={false}
          id="address-readonly-edit"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-99"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-127 edit-icon edit-icon-button"
            >
              edit
            </span>
            Edit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
